pkg/process/procutil/gpu_detector.go

```diff
@@ -0,0 +1,78 @@
+// Unless explicitly stated otherwise all files in this repository are licensed
+// under the Apache License Version 2.0.
+// This product includes software developed at Datadog (https://www.datadoghq.com/).
+// Copyright 2016-present Datadog, Inc.
+
+package procutil
+
+import (
+	"sync"
+
+	workloadmeta "github.com/DataDog/datadog-agent/comp/core/workloadmeta/def"
+	"github.com/DataDog/datadog-agent/pkg/util/log"
+)
+
+type GPUDetector struct {
+	DetectedGPU bool
+	gpuEventsCh chan workloadmeta.EventBundle
+	stopCh      chan struct{}
+	mu          sync.Mutex
+	wmeta       workloadmeta.Component
+}
+
+func NewGPUDetector(wmeta workloadmeta.Component) *GPUDetector {
+	return &GPUDetector{
+		stopCh: make(chan struct{}),
+		wmeta:  wmeta,
+	}
+}
+
+func (g *GPUDetector) Run() {
+	// TODO: ensure this is correct
+	filter := workloadmeta.NewFilterBuilder().
```

Comment:
Ensured this was correct by matching config for events sent by the collector to WLM store: https://github.com/DataDog/datadog-agent/pull/32109/files#diff-739a3320df37987b0114fdf0c00e0776dc531aff6cf2160a5c00685218e943b6R105